app.config.yaml

```diff
@@ -41,3 +41,5 @@ application:
             function: actions/csv/index.js
             web: true
             runtime: nodejs:18
+  hooks:
+    pre-app-build: ./hooks/pre-app-build.js
```

Comment:
Don't we want `pre-app-deploy`?

Reply:
> the user will have to wait for the build to complete before they will know that they are in a protected workspace.

That's okey. It should not happen often.

I'd agree with @lenaorobei `pre-app-deploy` would be a better chose cuz [POLA](https://en.wikipedia.org/wiki/Principle_of_least_astonishment)